src/client/rest/RESTMethods.js

```diff
@@ -18,6 +18,7 @@ class RESTMethods {
 
   login(token = this.rest.client.token) {
     return new Promise((resolve, reject) => {
+      if(typeof token !== 'string') return reject('invalid token');
```

Comment:
you should reject with the error message in constants, and following our style, there should be a space after the `if`